src/paths/shortest_paths.c

```diff
@@ -1279,3 +1280,315 @@ igraph_error_t igraph_diameter_dijkstra(const igraph_t *graph,
     }
     return IGRAPH_SUCCESS;
 }
+
+static igraph_error_t igraph_i_semidelete_vertex(
+    const igraph_t *graph, igraph_vector_t *weights,
+    igraph_integer_t vid, igraph_vector_int_t *edges_removed,
+    igraph_vector_int_t *eids
+) {
+    igraph_integer_t j, n;
+
+    IGRAPH_CHECK(igraph_incident(graph, eids, vid, IGRAPH_ALL));
+
+    n = igraph_vector_int_size(eids);
+    for (j = 0; j < n; j++) {
+        igraph_integer_t eid = VECTOR(*eids)[j];
+        IGRAPH_CHECK(igraph_vector_int_push_back(edges_removed, eid));
+        VECTOR(*weights)[eid] = IGRAPH_INFINITY;
+    }
+
+    return IGRAPH_SUCCESS;
+}
+
+static igraph_bool_t igraph_i_has_edge_with_infinite_weight(
+    const igraph_vector_int_t* path, const igraph_vector_t* weights
+) {
+    igraph_integer_t i, n;
+
+    n = weights ? igraph_vector_int_size(path) : 0;
+    for (i = 0; i < n; i++) {
+        igraph_integer_t edge = VECTOR(*path)[i];
+        if (!IGRAPH_FINITE(VECTOR(*weights)[edge])) {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+static igraph_real_t igraph_i_get_total_weight_of_path(
+    igraph_vector_int_t* path, const igraph_vector_t* weights
+) {
+    igraph_integer_t i, n = igraph_vector_int_size(path);
+    igraph_real_t result;
+
+    if (weights) {
+        result = 0;
+        for (i = 0; i < n; i++) {
+            igraph_integer_t edge = VECTOR(*path)[i];
+            result += VECTOR(*weights)[edge];
+        }
+    } else {
+        result = n;
+    }
+
+    return result;
+}
+
+/**
+ * \function igraph_get_k_shortest_paths
+ * \brief K shortest paths (geodesics) between two vertices.
```

Comment:
```suggestion
 * \brief k shortest paths between two vertices.
```

This function doesn't quite compute geodesics. Only the shortest path is a geodesic, not the 2nd shortest one.

Reply:
Isn't geodesic just a synonym for shortest path? The second shortest path is not a geodesic in the same way as the second shortest path is not a shortest path.

Reply:
I do think it's fine without :)